File: nicolify/frontend/src/features/abel/types/icp.ts
// cap: abel.icp-buyer
// story-origin: nicolify-r1-abel-icp-buyer T-FE-2
/**
 * icp.ts — TypeScript types for ICP entities (camelCase mirror of BE DTOs).
 *
 * Mirrors IcpResponse + IcpListItem DTOs from 03-arch-fe.md §6.
 * ISO 8601 datetimes as string. Optional fields explicit.
 *
 * Named exports only (NO default exports) per FSD-Lite enforce.
 * spec_anchor: 03-arch-fe.md §6 TypeScript Types
 */

export type IcpStatus = "borrador" | "listo";
export type IcpOrigin = "manual" | "draft";

export interface Icp {
  id: string;
  label: string;
  description: string | null;
  vertical: string | null;
  companySize: string | null;
  geo: string | null;
  businessModel: string | null;
  /** Decimal as string — preserved currency, never convert on write (RN-11) */
  avgTicket: string | null;
  /** ISO 4217 currency code — always paired with avgTicket (RN-11) */
  avgTicketCurrency: string | null;
  salesCycle: string | null;
  mainPain: string | null;
  salesAngle: string | null;
  signals: string[];
  antiPattern: string | null;
  status: IcpStatus;
  origin: IcpOrigin;
  /** Count of buyers under this ICP */
  buyerCount: number;
  createdAt: string | null; // ISO 8601
  updatedAt: string | null; // ISO 8601
}

export interface IcpListItem {
  id: string;
  label: string;
  vertical: string | null;
  status: IcpStatus;
  buyerCount: number;
}

export interface IcpCreatePayload {
  label: string;
  description?: string | null;
  vertical?: string | null;
  companySize?: string | null;
  geo?: string | null;
  businessModel?: string | null;
  avgTicket?: string | null;
  avgTicketCurrency?: string | null;
  salesCycle?: string | null;
  mainPain?: string | null;
  salesAngle?: string | null;
  signals?: string[];
  antiPattern?: string | null;
}

export type IcpPatchPayload = Partial<IcpCreatePayload>;

File: nicolify/backend/src/modules/nicolify/abel/domain/icp.py
# cap: abel/icp-buyer  # noqa: ERA001
"""ICP (Ideal Customer Profile) domain entity — brand-local, net-new B2B concept.

Account-level entity (empresa prospecto/cliente ideal). No engine equivalent.
Lift candidate: post-merge cuando saasora/inmoflow arranquen (N≥2 B2B brands).

Architecture decisions:
- IcpStatus: borrador (draft-first RN-2) → listo (mark-ready RN-8)
- IcpOrigin: manual vs draft (extraído por Abel T-AG-1)
- tenant_id raíz OBLIGATORIO (RN-1 tenant isolation)
- deleted_at soft delete OBLIGATORIO (no hard deletes)
- avg_ticket_currency preservado sin convertir (RN-11 currency-handling)
- signals[]: consumer Christian (outbound prospecting)
- consumer_agent fields: main_pain+sales_angle → Brenda+Christian

DDD: pure Python domain entity. CERO framework imports (SQLAlchemy, FastAPI).
"""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from enum import StrEnum
from uuid import UUID

from luana_core_platform.domain.base_entity import BaseEntity
from pydantic import Field


class IcpStatus(StrEnum):
    """Estado del ICP — draft-first (RN-2)."""

    BORRADOR = "borrador"
    LISTO = "listo"


class IcpOrigin(StrEnum):
    """Origen del ICP — manual vs extraído por Abel (T-AG-1)."""

    MANUAL = "manual"
    DRAFT = "draft"


class Icp(BaseEntity):
    """Ideal Customer Profile — entidad de cuenta B2B brand-local.

    Representa la empresa/vertical ideal que la agencia busca como cliente.
    Un ICP agrupa N Buyers (stakeholders de la empresa).

    Campos consumer_agent (RN-4 WhatForChip):
    - main_pain, sales_angle → consumer: Brenda (pauta) + Christian (outbound)
    - signals → consumer: Christian (prospecting triggers)
    - anti_pattern → consumer: Christian (calificación negativa)
    """

    id: UUID
    tenant_id: UUID  # RN-1 — raíz tenant isolation
    label: str  # RN-7 único por tenant (case-insensitive, soft-delete aware)
    description: str | None = None
    vertical: str | None = None  # RN-8 mínimo para mark-ready
    company_size: str | None = None
    geo: str | None = None
    business_model: str | None = None
    avg_ticket: Decimal | None = None  # RN-11 — no convertir on-write
    avg_ticket_currency: str | None = None  # RN-11 — preservar moneda original
    sales_cycle: str | None = None
    main_pain: str | None = None  # RN-8 mínimo · consumer: brenda + christian
    sales_angle: str | None = None  # RN-8 mínimo · consumer: brenda + christian
    signals: list[str] = Field(default_factory=list)  # consumer: christian
    anti_pattern: str | None = None  # consumer: christian (calificación negativa)
    status: IcpStatus = IcpStatus.BORRADOR  # RN-2 draft-first — siempre nace borrador
    origin: IcpOrigin = IcpOrigin.MANUAL  # se sobreescribe a DRAFT en extracción T-AG-1
    created_at: datetime | None = None
    updated_at: datetime | None = None
    deleted_at: datetime | None = None  # soft delete OBLIGATORIO — nunca hard delete
